src/main/java/com/google/jenkins/plugins/computeengine/client/ComputeClient.java

```diff
@@ -31,6 +33,9 @@
 public class ComputeClient {
     private Compute compute;
 
+    private static final Logger LOGGER = Logger.getLogger(ComputeClient.class.getName());
+    private static final long SNAPSHOT_TIMEOUT_MILLISECONDS = 600 * 1000;
```

Comment:
After reading this code Im really not sure how long we want to wait here :|

Maybe use something like this code below (If I read correctly what you are trying to set)
`TimeUnit.SECONDS.toMillis(600)`
Or even better 
`TimeUnit.MINUTES.toMillis(10);`

Reply:
Nice, good catch.